### PR TITLE
fix(plasma-ui): improve type-coverage

### PR DESCRIPTION
**Pull request by Yeti-or**
:baby_chick: Download canary assets:

[colors--canary.234.3542704878.xml](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/colors--canary.234.3542704878.xml)  
[color_sbermarket_ios-swift--canary.234.3542704878.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.234.3542704878.swift)  
[color_sbermarket_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.234.3542704878.kt)  
[color_sbermarket_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.234.3542704878.ts)  
[color_sberprime_ios-swift--canary.234.3542704878.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.234.3542704878.swift)  
[color_sberprime_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.234.3542704878.kt)  
[color_sberprime_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.234.3542704878.ts)  
[PlasmaTokensColor--canary.234.3542704878.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/PlasmaTokensColor--canary.234.3542704878.swift)  
[typo_mage_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.234.3542704878.kt)  
[typo_mage_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.234.3542704878.ts)  
[typo_plasma_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.234.3542704878.kt)  
[typo_plasma_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.234.3542704878.ts)  
[typo_ruler_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.234.3542704878.kt)  
[typo_ruler_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.234.3542704878.ts)  
[typo_sage_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.234.3542704878.kt)  
[typo_sage_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.234.3542704878.ts)  
[typo_sbermarket_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.234.3542704878.kt)  
[typo_sbermarket_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.234.3542704878.ts)  
[typo_soulmate_kotlin--canary.234.3542704878.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.234.3542704878.kt)  
[typo_soulmate_react-native--canary.234.3542704878.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.234.3542704878.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-temple@1.114.2-canary.234.3542704878.0
  npm install @salutejs/plasma-ui@1.147.2-canary.234.3542704878.0
  # or 
  yarn add @salutejs/plasma-temple@1.114.2-canary.234.3542704878.0
  yarn add @salutejs/plasma-ui@1.147.2-canary.234.3542704878.0
  ```
</details>

### CODE REVIEW

File: packages/plasma-ui/src/components/Pickers/DatePicker.tsx [hunk 1]
```diff
@@ -174,27 +174,27 @@ export const DatePicker = ({
     );
 
     const onYearChange = React.useCallback(
-        ({ value: y }) => {
+        ({ value: y }: PickerItem) => {
```
Comment (Yakutoc) [inline packages/plasma-ui/src/components/Pickers/DatePicker.tsx:177]: [Minor]

`Omit< PickerItem, 'value'> & { value: number }`

Может сделать override, что бы явно не приводить в коде ниже?

Comment (Yeti-or) [reply packages/plasma-ui/src/components/Pickers/DatePicker.tsx:178]: C разбегу не вышло: https://github.com/salute-developers/plasma/issues/242
если у тебя есть желание попробуй

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-234/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-234/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-234/
  ui storybook: http://plasma.sberdevices.ru/pr-234/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-234/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-234/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-234/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-234/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/temple-storybook/

  </details>

**Review by Yakutoc — APPROVED** [review]

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/demo-canvas-app-pr-234/
  <details>
    <summary>New documentation</summary>
    <br />

    http://plasma.fine.sberdevices.ru/pr/demo-canvas-app-pr-234/

  </details>

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr-234/
  ui storybook: http://plasma.sberdevices.ru/pr-234/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr-234/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr-234/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr-234/temple-storybook/
  <details>
    <summary>New documentation</summary>
    <br />

    website: http://plasma.fine.sberdevices.ru/pr/pr-234/
    ui storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/ui-storybook/
    web storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/web-storybook/
    b2c storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/b2c-storybook/
    temple storybook: http://plasma.fine.sberdevices.ru/pr/pr-234/temple-storybook/

  </details>

**Review by neretin-trike — APPROVED** [review]

**Pull request merged**